### PR TITLE
feat: add verifier for user ID (VF-2849)

### PR DESCRIPTION
**Pull request by trs**
**Fixes or implements VF-2849**

### Brief description. What is this change?

Instead of throwing an error during skill execution when the request body is missing the user ID, we can add a `Verifier` to check that the body contains the request body.

This will cause a 400 error to be returned, instead of a 500.

### CODE REVIEW

**Review by DecathectZero — CHANGES_REQUESTED** [review]
I don't think the underlying issue gets solved:

So sometimes Alexa sends this `CanfulfillIntentRequest` that INTENTIONALLY doesn't have a user Id, and this is expected behavior:
https://developer.amazon.com/en-US/docs/alexa/custom-skills/implement-canfulfillintentrequest-for-name-free-interaction.html#prepare-and-test-your-skill-for-certification

The idea is that Alexa sends this anonymous intent with no state really, and we have to tell it if our skill (app) is ready to accept this intent or not.
this is the expected response for it:
https://developer.amazon.com/en-US/docs/alexa/custom-skills/implement-canfulfillintentrequest-for-name-free-interaction.html#sample-intent-response

I think instead we could make a very specialized handler here that deals with it:
https://github.com/voiceflow/alexa-runtime/blob/c6c84238aee05fba35379cbf9d6da3fb562006de/lib/services/alexa/index.ts#L92-L103
Basically we can look at all the intents in the starting state and see if the requested intent matches them or not. Right now we have a `CAN_FULFILL_INTENT` with the Launch Handler but that's the wrong place to look. One problem might be this `RequestInterceptorGenerator` which requires the userId to look up the user's state.

Essentially we want to make a whole separate flow where we totally don't care about the user's state and only the voiceflow diagrams to compare this canfulfillrequest intent against.
